src/onikeiso/queue.c

```diff
@@ -0,0 +1,43 @@
+#include "queue.h"
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+void create_queue(Queue* queue, int max_capacity) {
+    queue->items = malloc(max_capacity * sizeof(void*));
+
+    queue->stored = 0;
+    queue->max_capacity = max_capacity;
+
+    for(int i = 0; i < max_capacity; ++i) {
+        queue->items[i] = NULL;
+    }
+}
+
+void queue_push(Queue* queue, void* item) {
+    if(queue->stored + 1 <= queue->max_capacity) {
+        queue->items[queue->stored] = item;
+
+        ++queue->stored;
+    }
```

Comment:
Maybe it is better to implement a queue as dynamic array (increase it's size dynamically), because `max_capacity` could be exceeded in real scenario.